### PR TITLE
move netty dependency from api to intermediate module

### PR DESCRIPTION
**Pull request by aharonha**
to make api dependency tree slimmer

### CODE REVIEW

File: services-transport-api/pom.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>io.scalecube</groupId>
+    <artifactId>scalecube-services-parent</artifactId>
+    <version>2.5.7-SNAPSHOT</version>
+  </parent>
+  <artifactId>services-transport-api</artifactId>
```
Comment (segabriel) [inline services-transport-api/pom.xml]: let's change it to `scalecube-services-transport-api` like other module names in this project.
IMHO it's not appropriate module name - classes which hosted here are simple helpers/utils to work with netty byte buffer.

**Comment by segabriel** [comment]
Please move `io.scalecube.services.codec.ServiceMessageCodecTest` inside the new module

File: services-transport-api/pom.xml (hunk 1 above)
Comment (aharonha) [reply services-transport-api/pom.xml]: actually it is `scalecube-services-transport-utils`. what do you think?

**Comment by aharonha** [comment]
> move `io.scalecube.services.codec.ServiceMessageCodecTest` inside the new module

@segabriel I Can not. there are some more dependencies on scalecube services here then just the netty.

File: services-transport-api/pom.xml (hunk 1 above)
Comment (aharonha) [reply services-transport-api/pom.xml]: Done in 0a23a9e

**Review by artem-v — APPROVED** [review]

**Comment by artem-v** [comment]
I propose to close this PR in favor of https://github.com/scalecube/scalecube-services/pull/556

**Review by ronenhamias — CHANGES_REQUESTED** [review]
tests are failing

**Comment by artem-v** [comment]
Closing in favour of https://github.com/scalecube/scalecube-services/pull/556

**Pull request closed without merging**